### ISSUE TITLE
libzfs_sendrecv.c: error: ‘clp’ may be used uninitialized in this function

### ISSUE THREAD

**Issue by kernelOfTruth** [zfsonlinux/zfs#4796]
Just encountered the following compilation error during kernel upgrade (with gcc 5.4.0):

```
libzfs_sendrecv.c: In function ‘zfs_receive’:
libzfs_sendrecv.c:3395:26: error: ‘clp’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
   if (zhp == NULL || clp == NULL || err)
                          ^
cc1: all warnings being treated as errors
make[3]: *** [Makefile:779: libzfs_sendrecv.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/var/tmp/portage/sys-fs/zfs-9999-r1/work/zfs-9999/lib/libzfs'
make[2]: *** [Makefile:642: all-recursive] Error 1
make[2]: Leaving directory '/var/tmp/portage/sys-fs/zfs-9999-r1/work/zfs-9999/lib'
make[1]: *** [Makefile:810: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/sys-fs/zfs-9999-r1/work/zfs-9999'
make: *** [Makefile:681: all] Error 2
 * ERROR: sys-fs/zfs-9999-r1::portage failed (compile phase):
 *   emake failed
```

Commits related to it are pretty old:

```
commit 172bb4bd5e4afef721dd4d2972d8680d983f144b
Author: Brian Behlendorf <behlendorf1@llnl.gov>
Date:   Thu Dec 11 11:08:09 2008 -0800

    Move the world out of /zfs/ and seperate out module build tree

commit b128c09fbee863d15be744a2ce602b514eddbe3a
Author: Brian Behlendorf <behlendorf1@llnl.gov>
Date:   Wed Dec 3 12:09:06 2008 -0800

    Rebase to OpenSolaris b103, in the process we are removing any code which did not originate from the OpenSolaris source.  These changes will be reintroduced in topic branches for easier tracking
```

ZoL:

https://github.com/zfsonlinux/zfs/blob/9f8026c802bf3ed23c66295ac4a0b07aa2cba528/lib/libzfs/libzfs_sendrecv.c#L3395

Illumos-Gate:

https://github.com/illumos/illumos-gate/blob/9adfa60d484ce2435f5af77cc99dcd4e692b6660/usr/src/lib/libzfs/common/libzfs_sendrecv.c#L3734

**Comment by behlendorf** [comment]
@kernelOfTruth this is a false positive flagged by gcc, due to how the code is laid out the comparison will never be done when clp is undefined.  However, I'm not at all opposed to a time patch which initialized clp to NULL to silence the warning if you want to open a PR.

**Comment by behlendorf** [comment]
Resolved by 689f093ebcfa0d57895495050d2b470ed2bef52e
